src/Agent.Worker/JobRunner.cs

```diff
@@ -279,6 +279,30 @@ public async Task<TaskResult> RunAsync(Pipelines.AgentJobRequestMessage message,
                     }
                 }
 
+#if OS_WINDOWS
+            bool retainDefaultEncoding = jobContext.Variables.GetBoolean(Constants.Variables.Agent.RetainDefaultEncoding) ?? false;
+            if (!retainDefaultEncoding)
+            {
+                using (var p = HostContext.CreateService<IProcessInvoker>())
+                {
+                    int exitCode = await p.ExecuteAsync(workingDirectory: HostContext.GetDirectory(WellKnownDirectory.Work),
+                                             fileName: WhichUtil.Which("chcp", true, Trace),
+                                             arguments: "65001",
+                                             environment: null,
+                                             cancellationToken: CancellationToken.None);
+                    if (exitCode == 0)
+                    {
+                        Trace.Info("Successfully changed to code page 65001 (UTF8)");
+                        Console.OutputEncoding = Encoding.UTF8;
```

Comment:
i think we might not need this. :D

Reply:
It looks like we do, when I get rid of it everything that was working stops (it doesn't fail, but special characters are garbage again). Still trying to figure out why